indico/modules/categories/client/js/calendar.js

```diff
@@ -74,19 +86,87 @@ import {$T} from 'indico/utils/i18n';
           }
         }
 
+        function setupCalendarLegend(data, items, legendContainer) {
+          const onFilterChanged = filterBy => {
+            groupBy = filterBy;
+            calendar.refetchEvents();
+          };
+          ReactDOM.render(
+            React.createElement(CalendarLegend, {items, groupBy: data.group_by, onFilterChanged}),
```

Comment:
you can use jsx syntax in .js files as well, so no need to manually calls `React.createElement()`

Reply:
Thanks for the tip! Fixed in [c16abfa](https://github.com/indico/indico/pull/6110/commits/c16abfad4e60f93ed7c3db25c0f5a84fbdc402f0).